qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -150,30 +151,38 @@ def result(self, timeout=None, wait=5):
         return this_result
 
     def cancel(self):
-        """Attempt to cancel job. Currently this is only possible on
-        commercial systems.
+        """Attempt to cancel a job. If it is not possible, check the
+        ```exception``` property for more information.
+
         Returns:
-            bool: True if job can be cancelled, else False.
+            bool: True if job can be cancelled, else False. Currently this is
+            only possible on commercial systems.
 
         Raises:
             IBMQJobError: if server returned error
         """
-        if self._is_commercial:
-            hub = self._api.config['hub']
-            group = self._api.config['group']
-            project = self._api.config['project']
+        hub = self._api.config.get('hub', None)
+        group = self._api.config.get('group', None)
+        project = self._api.config.get('project', None)
+
+        cancelled = False
+        try:
             response = self._api.cancel_job(self._id, hub, group, project)
-            if 'error' in response:
+            errored = 'error' in response
+            cancelled = not errored
```

Comment:
just a comment... why not:
```python
self._cancelled = not errored
```

Reply:
Could be. No preferences. Can change if you want.

Reply:
I prefer it this way, actually. It ensures the state of the object is modified in only one place.